sources/test/user/data.test.js

```diff
@@ -0,0 +1,195 @@
+const {beforeTests,afterTests, apolloServer, PERMISSION_DENIED_TEST} = require("../testHelper");
+
+// Pre and Post Test Scripts
+beforeAll(beforeTests);
+afterAll(afterTests);
+
+describe('Results: Users Queries and Mutations', () => {  
+
+  const { query, mutate } = apolloServer("a8mjiKYtt0PefnS524",["users.all","users.Auth","users.byName","users.byId","users.Update","users.Delete"]);
+  let testUser={
+    name:"Harish",
+    username:"HarishTeens"  
+  };
+  it('Get all Users', async () => {    
+      const GET_USERS = `
+        {
+          users{
+            ... on User{
+              name
+            }
+          }
+    
+        }
+      `;
+
+      const response = await query({ query: GET_USERS });
+      expect(response.data.users).toEqual([]);
+    });
+
+  it("Authenticate User(Sign up)",async ()=>{
+    const AUTH_USER = `
+      mutation {
+        authUser(user:{
+          username:"`+testUser.username+`",
+          gmailAuthMail:"arishh2@gmail.com",
+          name:"`+testUser.name+`"
+        }){
+          ... on User{
+            name
+            id,
+            username
+          }
+        }
+      }
+    `;
+
+    const response = await mutate({ mutation: AUTH_USER });
+    expect(response.errors).toEqual(undefined)
+    const userResponse= response.data.authUser;
+    testUser.id=userResponse.id;
+    expect(userResponse).toEqual(testUser);
+  })
+
+  it("Authenticate User(Sign in) + Level 1 check",async ()=>{
+    const testUser4={
+      ...testUser,
+      clubAccess:[{
+        level:"1",
+        name:testUser.name
+      }]
+    }
+    const AUTH_USER = `
+      mutation {
+        authUser{
+          ... on User{
+            name
+            id,
+            username,
+            clubAccess{
+              level,
+              name
+            }
+          }
+        }
+      }
+    `;
+
+    const response = await mutate({ mutation: AUTH_USER });
+    expect(response.errors).toEqual(undefined)
+    const userResponse= response.data.authUser;
+    expect(userResponse).toEqual(testUser4);
+  })
+
+  it('user by username', async () => {    
+    const GET_USER = `
+      {
+        userByUsername(username:"`+testUser.username+`"){            
+          ... on User{
+            name
+            id,
+            username
+          }
+        } 
+      }
+    `;
+
+    const response = await query({ query: GET_USER });
+    expect(response.data.userByUsername).toEqual(testUser);
+  }); 
```

Comment:
This test relies on addUserTest passing. Not an ideal strategy I suppose?

Reply:
I need to work on making these tests completely independent from each other. Presently each block is isolated